elliottlib/cli/verify_attached_bugs_cli.py

```diff
@@ -0,0 +1,122 @@
+import click
+from kerberos import GSSError
+
+from errata_tool import Erratum
+
+from elliottlib import bzutil, errata
+from elliottlib.cli.common import cli, pass_runtime
+from elliottlib.exceptions import ElliottFatalError
+from elliottlib.util import (exit_unauthenticated, red_print, green_print, minor_version_tuple)
+
+
+@cli.command("verify-attached-bugs", short_help="Verify bugs in a release will not be regressed in the next version")
+@click.argument("advisories", nargs=-1, type=click.IntRange(1), required=False)
+@pass_runtime
+def verify_attached_bugs_cli(runtime, advisories):
+    """
+    Verify the bugs in the advisories (specified as arguments or in group.yml) for a release.
+    Requires a runtime to ensure that all bugs in the advisories match the runtime version.
+    Also ensures that bugs in the next release which block bugs in these advisories have
+    been verified, as those represent backports we do not want to regress in upgrades.
+
+    If any verification fails, a text explanation is given and the return code is 1.
+    Otherwise, prints the number of bugs in the advisories and exits with success.
+    """
+    runtime.initialize()
+    advisories = advisories or [a for a in runtime.group_config.get('advisories', {}).values()]
+    if not advisories:
+        red_print("No advisories specified on command line or in group.yml")
+        exit(1)
+    BugValidator(runtime).validate(advisories)
+
+
+class BugValidator:
+
+    def __init__(self, runtime):
+        self.runtime = runtime
+        self.bz_data = runtime.gitdata.load_data(key='bugzilla').data
+        self.target_releases = self.bz_data['target_release']
+        self.product = self.bz_data['product']
+        self.bzapi = bzutil.get_bzapi(self.bz_data)
+        self.problems = []
+
+    def validate(self, advisories):
+        bugs = self._get_attached_filtered_bugs(advisories)
+        blocking_bugs_for = self._get_blocking_bugs_for(bugs)
+        self._verify_blocking_bugs(blocking_bugs_for)
+        if self.problems:
+            red_print("Some bug problems were listed above. Please investigate.")
+            exit(1)
+        green_print("All bugs were verified.")
+
+    def _get_attached_bugs(self, advisories):
+        # get bugs attached to all advisories
+        bugs = []
+        try:
+            for advisory in advisories:
+                green_print(f"Retrieving bugs for advisory {advisory}")
+                bugs.extend(errata.get_bug_ids(advisory))
+        except GSSError:
+            exit_unauthenticated()
+        green_print(f"Found {len(bugs)} bugs")
```

Comment:
Though it is not required, I think it is better to remove duplicate bugs here (it is possible that a flaw bug will be attached to multiple advisories):
```suggestion
        bugs = list(set(bugs))
        green_print(f"Found {len(bugs)} bugs")
```

Reply:
good idea, but i just changed it to be a set from the start instead :)